packages/@lwc/engine-core/src/framework/modules/props.ts

```diff
@@ -24,13 +24,6 @@ function update(oldVnode: VElement, vnode: VElement) {
         return;
     }
 
-    if (process.env.NODE_ENV !== 'production') {
```

Comment:
Ok, I think it is ok to remove this invariant since the user has no saying on it. This was just added because in the initial iteration of these modules, the user could provide their own vnodes. So, I think it is fine.